### ISSUE TITLE
Segmentation Fault (core dumped) Mongodb 3.2.3

### ISSUE THREAD

**Issue by lporras** [joyent/smartos-live#611]
Hey guys,

I installed mongodb 3.2.3 on a machine that previously had mongodb 3.0.7, and when I try to connect to the mongo server I get this:

```
# mongo --port 27018
MongoDB shell version: 3.2.3
Segmentation Fault (core dumped)
```
I am sure I do not have another mongodb process running,  I removed mongodb 3.0.7 before installing the new version.

PD: The reason I'm trying to upgrade to mongodb 3.2.3 is because MMS is indicating this: All MongoDB deployments using the WiredTiger storage engine should update to MongoDB 3.0.8 immediately. Please see [SERVER-21275](https://jira.mongodb.org/browse/SERVER-21275) for details.

I have this kind of machine: g3-standard-8-smartos

**Comment by jclulow** [comment]
How are you installing the updated MongoDB software?

**Comment by lporras** [comment]
hey @jclulow via pkgin: ```pkgin install mongodb-3.2.3```

**Comment by rmustacc** [comment]
@mamash didn't you have an upstream bug against Mongo for something similar?

**Comment by mamash** [comment]
Yup, https://jira.mongodb.org/browse/SERVER-22927

**Comment by mamash** [comment]
Not even the official 3.2 binaries work on a standard SunOS kernel, they only work on the i86xpv (Xen) kernel. This is being worked on by upstream.

**Comment by lporras** [comment]
@mamash @jclulow Thanks for the information.  Have you tried installing Mongo 3.0.8 using binaries?

**Comment by mamash** [comment]
I'm already working on bringing the latest 3.0.11 back into 2015Q4 as an interim workaround.

**Comment by lporras** [comment]
nice @mamash and thank you again 👍

**Comment by mamash** [comment]
3.0.11 is in 2015Q4 as of last night. I'll update the 3.2 package as soon as upstream releases a version with a fix for the mozjs segfault.

**Comment by Fr0stoff** [comment]
I success run 3.2.3 on SmartOS, just mongo client core dumped on connect to server,
when I connect from remote client on some Linux everything work as expected.
I think the problem in libstdc++.so.6  which should be provided from here http://downloads.mongodb.org.s3.amazonaws.com/sunos5/mongo-extra-64.tgz 
but last time is unavailable.

**Comment by bastianwegge** [comment]
Hint: You can use port-forwarding to access a remote mongodb with your local client as a workaround.

```
ssh -fN -l root -i ~/.ssh/id_rsa -L 9999:localhost:27017 host.com
```

- 9999 is your local port
- 27017 is the remote port
- host.com should be your host

**Comment by teotikalki** [comment]
Is there any progress on this? I find myself unhappily here after initializing a mongodb zone during my setup of a nodebb instance. Upgrading my mongodb to 3.2.3 rendered my unable to connect locally: [root@mongodb ~]# mongo
MongoDB shell version: 3.2.3
Segmentation Fault (core dumped)
@Fr0stoff was correct in that I was able to connect remotely (mine was from a second smartos native zone; I used the latest nodejs native dataset as the base for nodebb) and nodebb worked... HOWEVER, this is only true because I did the user/database creation before I did the upgrade. Now I have a server which I can't interact with since it seems that I need to locally add the root user according to something like: http://stackoverflow.com/questions/23943651/mongodb-admin-user-not-authorized and I didn't do this before I upgraded to 3.2.3.
I'm currently thinking of making a new mongodb zone, adding the root user, connecting remotely, and THEN upgrading, but re-doing everything like that is... bad. I actually feel that having a package in the repository that breaks a server is just... BAD. If mongo 3.2.3 doesn't work (properly) on SmartOS then it shouldn't be what 'pkgin upgrade' does!

**Comment by mamash** [comment]
Sadly nothing has changed on this issue since my [last comment](https://github.com/joyent/pkgsrc/issues/389#issuecomment-233626086) on the Joyent pkgsrc issue page. Upstream only fixed this on the 3.4 branch, which we can't backport because of the strict GCC 5 requirement. As far as I can see, the entire 3.2 branch is broked in this regard.
